packages/field-base/src/label-controller.js

```diff
@@ -0,0 +1,208 @@
+/**
```

Comment:
Tried adding a label dynamically with:
```js
const lbl = document.createElement('label');
lbl.textContent = 'new label';
lbl.slot = 'label';
checkbox.append(lbl);
```

Before:
<img width="662" alt="Screenshot 2021-12-22 at 11 35 38" src="https://user-images.githubusercontent.com/1222264/147071123-7071d1c5-fd73-4e5d-af92-3ea32d74f223.png">

After:
<img width="663" alt="Screenshot 2021-12-22 at 11 35 17" src="https://user-images.githubusercontent.com/1222264/147071164-1c130ae4-1dd2-4efc-be18-774ecf004eb9.png">

Seems that the label is replaced but the new label is missing the `for` attribute

Reply:
Looks like I will have to make `SlotController` extend `EventTarget` and dispatch events on label change instead of using `labelChangedCallback`, as we need to notify both `LabeleldInputController` and `FieldController`.

Reply:
Done.